### PR TITLE
contract wrapper in python

### PR DESCRIPTION
**Pull request by hiutin0**
1. add Class Contract to do the following things 
1) show contract function
2) get contract info and content
3) get token balance and info
4) register contract and execute contract
5) contract permitted (with split function and without split function)

2. add contract translator to deserialize a contract

3. add contract build to serialize a contract (with default case in contract permitted and user defined case)

4. add contract opcode to denote the input parameter of opc function

5. add contract meta to store the parameter of a default contract

6. add contract deser

7. add three error types

8. add dataentry Class

9. add two functions in account.py (get_tx_status and get_tx_height)

10. write a sample test for the project

### CODE REVIEW

File: account.py [hunk 1]
```diff
@@ -457,3 +457,29 @@ def check_node(self, other_node_host=None):
             res = self.chain.self_check()
         # add more check if need
         return res
+
+    def get_tx_status(self, tx_id):
+        if is_offline():
```
[the remark this answers is not in the document]
Comment (zl730) [reply account.py]: May use decorator here

File: contract.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        else:
+            logging.debug(resp)
+            return resp.get('info')
+
+    def get_contract_content(self, wrapper, contract_id):
+        resp = wrapper.request('/contract/content/%s' % (contract_id))
+        logging.debug(resp)
+        return resp
+
+    def get_token_balance(self, wrapper, address, contract_id, token_index):
```
[the remark this answers is not in the document]
Comment (Icermli) [reply contract.py]: Separating into two functions is better

File: account.py [hunk 3]
```diff
@@ -457,3 +457,29 @@ def check_node(self, other_node_host=None):
...
+            pyvsystems.throw_error("Cannot check transaction in offline mode.", NetworkException)
+        utx_res = self.chain.unconfirmed_tx(tx_id)
+        if "id" in utx_res:
+            pyvsystems.throw_error("Transaction {} is pending in UTX pool.".format(tx_id), InvalidStatus)
+        else:
+            tx_res = self.chain.tx(tx_id)
+            if 'id' not in tx_res:
+                pyvsystems.throw_error("Transaction does not exist!", InvalidStatus)
+            else:
+                return tx_res['height']
```
Comment (Icermli) [inline account.py]: Maybe should return None if not status

File: account.py [hunk 4]
```diff
@@ -457,3 +457,29 @@ def check_node(self, other_node_host=None):
...
+            pyvsystems.throw_error("Cannot check transaction in offline mode.", NetworkException)
+        utx_res = self.chain.unconfirmed_tx(tx_id)
+        if "id" in utx_res:
+            pyvsystems.throw_error("Transaction {} is pending in UTX pool.".format(tx_id), InvalidStatus)
+        else:
+            tx_res = self.chain.tx(tx_id)
+            if 'id' not in tx_res:
+                pyvsystems.throw_error("Transaction does not exist!", InvalidStatus)
+            else:
+                return tx_res['status']
```
Comment (Icermli) [inline account.py]: Maybe should return None if not status

Comment (hiutin0) [reply account.py]: add a get_tx_function in update and return None if not status

File: account.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hiutin0) [reply account.py]: add a chekc_is_offline function

File: account.py [hunk 5]
```diff
@@ -457,3 +457,29 @@ def check_node(self, other_node_host=None):
             res = self.chain.self_check()
         # add more check if need
         return res
+
+    def get_tx_status(self, tx_id):
+        if is_offline():
+            pyvsystems.throw_error("Cannot check transaction in offline mode.", NetworkException)
+        utx_res = self.chain.unconfirmed_tx(tx_id)
+        if "id" in utx_res:
```
[the remark this answers is not in the document]
Comment (hiutin0) [reply account.py]: add a function of check_tx_is_unconfirmed()

**Pull request merged**